gridstatus/pjm.py

```diff
@@ -3012,8 +3013,306 @@ def get_projected_peak_tie_flow(
             [
                 "Interval Start",
                 "Interval End",
-                "Generated At",
+                "Publish Time",
                 "Interface",
                 "Scheduled Tie Flow",
             ]
         ]
+
+    @support_date_range(frequency=None)
+    def get_actual_operational_statistics(
+        self,
+        date: str | pd.Timestamp,
+        end: str | pd.Timestamp | None = None,
+        verbose: bool = False,
+    ) -> pd.DataFrame:
+        """
+        Retrieves the actual operational statistics data from:
+        https://dataminer2.pjm.com/feed/ops_sum_prev_period/definition
+        """
+        if date == "latest":
+            return self.get_actual_operational_statistics("today")
+
+        df = self._get_pjm_json(
+            "ops_sum_prev_period",
+            start=date,
+            end=end,
+            params={
+                "fields": "datetime_beginning_utc,datetime_ending_utc,generated_at_ept,area,area_load_forecast,actual_load,dispatch_rate",
+            },
+            filter_timestamp_name="generated_at",
+            verbose=verbose,
+        )
+
+        df = df.rename(
+            columns={
+                "datetime_beginning_utc": "Interval Start",
+                "datetime_ending_utc": "Interval End",
+                "generated_at_ept": "Publish Time",
+                "area": "Area",
+                "area_load_forecast": "Area Load Forecast",
+                "actual_load": "Actual Load",
+                "dispatch_rate": "Dispatch Rate",
+            },
+        )
+
+        return (
+            df[
+                [
+                    "Interval Start",
+                    "Interval End",
+                    "Publish Time",
+                    "Area",
+                    "Area Load Forecast",
+                    "Actual Load",
+                    "Dispatch Rate",
+                ]
+            ]
+            .sort_values(["Interval Start", "Area"])
+            .reset_index(drop=True)
+        )
+
+    def _filter_active_records(
+        self,
+        df: pd.DataFrame,
+        as_of: pd.Timestamp | None,
+    ) -> pd.DataFrame:
+        """Filter out records that are terminated before the given date"""
+        df["Effective Date"] = pd.to_datetime(
+            df["Effective Date"],
+            errors="coerce",
+        ).dt.tz_localize(
+            self.default_timezone,
+        )
+
+        df["Termination Date"] = df["Termination Date"].replace(
+            "9999-12-31T00:00:00",
+            None,
+        )
+        df["Termination Date"] = pd.to_datetime(
+            df["Termination Date"],
+            errors="coerce",
+        ).dt.tz_localize(
+            self.default_timezone,
+        )
+
+        if as_of is None:
+            return df
```

Comment:
Why would we call this method is `as_of` is `None`? Wouldn't that mean no filtering is done?